### PR TITLE
CT-2912 Amend case creation steps

### PR DESCRIPTION
**Pull request by Polsonby**
## Description
Plumbing work for subsequent tickets to add new steps and fields to the Offender SAR case creation flow. 

New flow: `[subject_details_step, requester_details_step, recipient_details_step, requested_info_step, request_details_step, date_received_step]`

This PR adds two new (empty) steps `recipient_details` and `request_details` to the flow as placeholders and ensures that the feature specs still work. 

Subsequent tickets must complete the Site Prism page object descriptions and method to fill in the fields

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots

image

image

### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2912

### Deployment
nope

### Manual testing instructions
Sign in as Brian Rix and step through the case creation flow for Offender SAR. There should be two new steps entitled "Where should the subject access request be sent?" and "What is the date on the request?"

### CODE REVIEW

File: spec/site_prism/page_objects/pages/application.rb [hunk 1]
```diff
@@ -35,7 +35,9 @@ def app_sections
         cases_new_sar:                  'Cases::New::SARPage',
         cases_new_offender_sar_subject_details:                 'Cases::New::OffenderSARPageSubjectDetails',
         cases_new_offender_sar_requester_details:                 'Cases::New::OffenderSARPageRequesterDetails',
+        cases_new_offender_sar_recipient_details:                 'Cases::New::OffenderSARPageRecipientDetails',
         cases_new_offender_sar_requested_info:                 'Cases::New::OffenderSARPageRequestedInfo',
+        cases_new_offender_sar_request_details:                 'Cases::New::OffenderSARPageRequestDetails',
```
Comment (willmcb) [inline spec/site_prism/page_objects/pages/application.rb:40]: Could this be more alphabetical? Small thing no reason to change if you don't think its necessary 😄

**Review by willmcb — APPROVED** [review]

File: app/form_models/offender_sar_case_form.rb [hunk 2]
```diff
@@ -52,7 +52,7 @@ def initialize(session)
 
   # @todo: Should these steps be defined in 'Steppable' or the controller
   def steps
-    %w[subject-details requester-details requested-info date-received].freeze
+    %w[subject-details requester-details recipient-details requested-info request-details date-received].freeze
```
[the remark this answers is not in the document]
Comment (Polsonby) [reply app/form_models/offender_sar_case_form.rb:55]: Cool, good suggestion, thanks! 

Yes, this is the order of the steps

File: config/locales/en.yml [hunk 3]
```diff
@@ -901,7 +901,9 @@ en:
     offender_sar:
       subject_heading: Who is the subject?
       requester_heading: Who is making the subject access request?
+      recipient_details_heading: Where should the subject access request be sent?
       requested_info_heading: Is there other information relevant for this request?
+      request_details_heading: What is the date on the request?
```
[the remark this answers is not in the document]
Comment (Polsonby) [reply config/locales/en.yml:906]: Totes agree, thanks for mentioning :) 

In this case it's just a starting point - we will be fine tuning the copy a few times I expect, but I've asked Alex if she has any tweaks before we merge https://mojdt.slack.com/archives/C581VE4QZ/p1593444080013000

File: spec/site_prism/page_objects/pages/application.rb (hunk 1 above)
Comment (Polsonby) [reply spec/site_prism/page_objects/pages/application.rb:40]: Good point - we try to keep things alphabetical everywhere else, will change 👍

**Pull request merged**